extmod/vfs_reader.c

```diff
@@ -34,21 +34,39 @@
 
 #if MICROPY_READER_VFS
 
+
 typedef struct _mp_reader_vfs_t {
     mp_obj_t file;
     uint16_t len;
     uint16_t pos;
-    byte buf[24];
+    uint8_t bufsize;
+    byte buf[];
 } mp_reader_vfs_t;
 
 STATIC mp_uint_t mp_reader_vfs_readbyte(void *data) {
     mp_reader_vfs_t *reader = (mp_reader_vfs_t *)data;
+    if (!reader->bufsize) {
```

Comment:
Instead of a different code path (adding code size), could we have the buffer size be always at least 1.

So if the ioctl returns zero, then the effective buffer size is 1, and we use the `mp_reader_vfs_t`'s `buf` in place of the `byte c` used here.

Because the mp_reader_vfs_t is always heap allocated, really what we should probably do is make the minimum buffer `sizeof(block)-9` (i.e. 7), otherwise those bytes are just wasted. And even for low-latency filesystems, it's still faster to buffer than not.

Reply:
I don't think we should support zero buffer size.  At least use 7 as the minimum since the bytes are anyway there.

For the default, I think it should be 23 so we retain the existing performance for filesystems like FAT and littlefs.  So if the ioctl is not supported it defaults to 23.  And the minimum value is 7, maximum is 255.